File: src/main/java/org/prog5/model/Location.java
package org.prog5.model;

import java.time.LocalDate;

public class Location {
    private final ObjetLocationnable objet;
    private final LocalDate date;

    public Location(ObjetLocationnable objet, LocalDate date) {
        if (objet == null || date == null) {
            throw new IllegalArgumentException("L'objet ou la date ne peut pas être nul");
        }
        this.objet = objet;
        this.date = date;
    }

    public ObjetLocationnable obtenirObjet() {
        return objet;
    }

    public LocalDate obtenirDate() {
        return date;
    }
}

File: version_javascript/model/location.js
import { ObjetLocationnable } from '../model/objet_locationnable.js';

export class Location {
  #objet;
  #date;

  constructor(objet, date) {
    if (!(objet instanceof ObjetLocationnable) || !date) {
      throw new Error("L'objet ou la date ne peut pas être nul");
    }
    this.#objet = objet;
    this.#date = date;
  }

  get objet() {
    return this.#objet;
  }

  get date() {
    return this.#date;
  }
}